src/lib/components/Editor/ContentPreviewContainer.tsx

```diff
@@ -0,0 +1,105 @@
+import { Grid, TextField, Typography } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import PostPreviewCard from '../PostPreview/PostPreviewCard';
+import { mockUser } from './mock/mockUser';
+
+export interface IContentPreviewContainerProps {
+  previewText: string;
+}
+
+const MAX_LENGTH = 150;
+
+const trunkLength = (str: string) => {
+  if (str.length > MAX_LENGTH) {
+    return str.slice(0, MAX_LENGTH);
+  }
+  return str;
+};
+
+const ContentPreviewContainer: React.FC<IContentPreviewContainerProps> = ({
+  previewText,
+}) => {
+  const [textFieldValue, setTextFieldValue] = useState<string>('');
+  const [isTextFieldManualyChanged, setisTextFieldManualyChanged] = useState<
+    boolean
+  >(false);
+  const [isPreviewValid, setIsPreviewValid] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (!isTextFieldManualyChanged) {
+      setTextFieldValue(trunkLength(previewText));
+    }
+  }, [textFieldValue, previewText]);
+
+  const onChangeHandler = (
+    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+  ) => {
+    setisTextFieldManualyChanged(true);
+    setTextFieldValue(e.target.value);
+    setIsPreviewValid(true);
+  };
+
+  const onKeyPressHandler = () => {
+    if (textFieldValue.length === MAX_LENGTH) {
+      setIsPreviewValid(false);
+    }
```

Comment:
you can do this check in `onChangeHandler `
no need for KeyPress

Reply:
fixed